ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1962,6 +2068,16 @@ IopCreateDeviceInstancePath(
         ASSERT(IoStatusBlock.Information == 0);
     }
 
+    if (IoStatusBlock.Information)
+    {
+        IsValidID = IopValidateID((PWCHAR)IoStatusBlock.Information, BusQueryInstanceID);
+
+        if (!IsValidID)
+        {
+            DPRINT1("Invalid InstanceID. DeviceNode - %p\n", DeviceNode);
```

Comment:
I suppose you should do more than that, either returning FALSE here, or bugchecking.

Reply:
I have already added "// FIXME logging" in the IopValidateID() function, which means using IoWriteErrorLogEntry() to write.
I tested (usbhub_new) in Windows - bugchecking does not happen. If I add an error (except for a spaces) - a flash drive attached to the hub is not recognized. But Windows continues to work.